test/suites/stability/lifecycle.robot

```diff
@@ -0,0 +1,74 @@
+*** Settings ***
+Documentation       Tests for MicroShift service lifecycle
+
+Resource            ../../resources/common.resource
+Resource            ../../resources/microshift-process.resource
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           restart
+
+
+*** Variables ***
+${RESTART_ATTEMPTS}     30
+
+
+*** Test Cases ***
+Restarting MicroShift In The Middle Of Startup Should Succeed
+    [Documentation]    Checks if restarting MicroShift during startup fails.
+    ...    (for example due to not stopped microshift-etcd.scope).
+
+    FOR    ${attempt}    IN RANGE    ${RESTART_ATTEMPTS}
+        ${time_to_sleep}=    Evaluate    random.sample(range(1, 40), 1)[0]    random
+        Log To Console    Attempt ${attempt} with ${time_to_sleep}s sleep
+        Stop MicroShift
+        Start MicroShift Without Blocking
+        Wait Until MicroShiftEtcd Is Active
+        # microshift-etcd.scope is running so microshift is mid startup
+        Sleep    ${time_to_sleep}s
+        Restart MicroShift Without Retry
+    END
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig    # for readiness checks
+
+Teardown
+    [Documentation]    Test suite teardown
+    Restart MicroShift
+    Logout MicroShift Host
+    Remove Kubeconfig
+
+Start MicroShift Without Blocking
```

Comment:
Do you think at least some of these keywords should be moved to the library?

Reply:
Done